packages/clerk-js/src/core/auth/SessionCookieService.ts

```diff
@@ -0,0 +1,134 @@
+import { setDevBrowserJWTInURL } from '@clerk/shared/devBrowser';
+import { is4xxError, isClerkAPIResponseError, isNetworkError } from '@clerk/shared/error';
+import type { Clerk, EnvironmentResource } from '@clerk/types';
+
+import { clerkCoreErrorTokenRefreshFailed, clerkMissingDevBrowserJwt } from '../errors';
+import { eventBus, events } from '../events';
+import type { FapiClient } from '../fapiClient';
+import type { ClientUatCookieHandler } from './cookies/clientUat';
+import { createClientUatCookie } from './cookies/clientUat';
+import type { SessionCookieHandler } from './cookies/session';
+import { createSessionCookie } from './cookies/session';
+import type { DevBrowser } from './devBrowser';
+import { createDevBrowser } from './devBrowser';
+import { SessionCookiePoller } from './SessionCookiePoller';
+
+// TODO: make SessionCookieService singleton since it handles updating cookies using a poller
+// and we need to avoid updating them concurrently.
+export class SessionCookieService {
```

Comment:
Can we provide a brief JSDoc comment here outlining what this class is responsible for? I believe it has expanded a bit with this PR. Maybe obvious now, but helpful for the future if more folks need to contribute.

Reply:
https://github.com/clerk/javascript/pull/3362#issuecomment-2110029255